iModelCore/ECPresentation/Source/Serialization/DefaultECPresentationSerializer.cpp

```diff
@@ -607,7 +607,7 @@ rapidjson::Document DefaultECPresentationSerializer::_AsJson(ContextR ctx, Conte
             rapidjson::Value nestedValuesJson(rapidjson::kArrayType);
             for (auto const& nestedItem : nestedContentEntry.second)
                 {
-                static const int valueSerializationFlags = ContentSetItem::SERIALIZE_PrimaryKeys | ContentSetItem::SERIALIZE_Values | ContentSetItem::SERIALIZE_DisplayValues | ContentSetItem::SERIALIZE_MergedFieldNames;
+                static const int valueSerializationFlags = ContentSetItem::SERIALIZE_PrimaryKeys | ContentSetItem::SERIALIZE_Values | ContentSetItem::SERIALIZE_DisplayValues | ContentSetItem::SERIALIZE_MergedFieldNames | ContentSetItem::SERIALIZE_DisplayLabel;

```

Comment:
I don't see a similar change in `IModelJsECPresentationSerializer` - is it not necessary there?

Reply:
You are right. Deeply nested content doesn't serialize properly. Fixed and added more tests from the core side